internal/consensus/reactor.go

```diff
@@ -411,10 +414,30 @@ func makeRoundStepMessage(rs *cstypes.RoundState) *tmcons.NewRoundStep {
 func (r *Reactor) sendNewRoundStepMessage(ctx context.Context, peerID types.NodeID) error {
 	return r.stateCh.Send(ctx, p2p.Envelope{
 		To:      peerID,
-		Message: makeRoundStepMessage(r.state.GetRoundState()),
+		Message: makeRoundStepMessage(r.getRoundState()),
 	})
 }
 
+func (r *Reactor) updateRoundStateRoutine() {
+	t := time.NewTicker(100 * time.Microsecond)
+	defer t.Stop()
+	for range t.C {
+		if !r.IsRunning() {
+			return
+		}
+		rs := r.getRoundState()
+		r.mtx.Lock()
+		r.rs = rs
+		r.mtx.Unlock()
+	}
+}
+
+func (conR *Reactor) getRoundState() *cstypes.RoundState {
```

Comment:
I think we already have a [`GetRoundState` method](https://github.com/tendermint/tendermint/blob/master/internal/consensus/state.go#L293
) that does this—can we get rid of that one? (Or use it instead of this—as I think getting rid of it would require renaming all the existing uses)

Reply:
So, this method operates on the Reactor's _copy_ of the `RoundState` instead of the `State`'s copy. The `State` copy is really the source of truth, this is just the gossip routine's cached version.

Reply:
Ohh, right, I should have looked more carefully at which receiver. 🙄